crates/sui-move-build/src/lib.rs

```diff
@@ -269,6 +272,52 @@ pub fn build_from_resolution_graph(
     })
 }
 
+pub fn resolve_toolchain_version(
+    lock_file: Option<PathBuf>,
+    default_flavor: Option<Flavor>,
+    default_edition: Option<Edition>,
+) -> anyhow::Result<ToolchainVersion> {
+    // Top-level invariant: if a flavor is set (i.e., default_flavor is Some) it must be valid.
+    match default_flavor {
+        Some(flavor) if flavor != Flavor::Sui => anyhow::bail!(
+            "The flavor of the Move compiler cannot be overridden with anything but \
+                        \"{}\", but the default override was set to: \"{flavor}\" in Move.toml",
+            Flavor::Sui
+        ),
+        _ => (),
+    };
+
+    let base_toolchain_version = ToolchainVersion {
+        compiler_version: env!("CARGO_PKG_VERSION").into(),
+        edition: default_edition.unwrap_or_default(),
+        flavor: default_flavor.unwrap_or(Flavor::Sui),
+    };
+
+    match lock_file {
+        Some(lock_file) if lock_file.exists() => {
+            let mut lock_path = File::open(lock_file)?;
+            let toolchain_version = lock_file::schema::ToolchainVersion::read(&mut lock_path)
+                .map_err(|e| anyhow!("Move.lock: {:?}", e))?;
+
+            if let Some(mut toolchain_version) = toolchain_version {
+                // Lock file specifies toolchain version. But: honor existing values if Some.
+                if let Some(edition) = default_edition {
+                    toolchain_version.edition = edition
+                }
+                if let Some(flavor) = default_flavor {
+                    toolchain_version.flavor = flavor
+                }
```

Comment:
The function contract here is that if `edition` and `flavor` are already set by the time we try to resolve them from the lock file, then they should take precedence (e.g., if the values originate from the `Move.toml`, we should force those options rather than `Move.lock` contents). Which, I think is the behavior we want? CC @tnowacki 

I'm not sure if we _actually_ set these values via `Move.toml` yet or rely on hardcoded defaults. From what I've seen I think we only hardcode defaults, but I think this function contract makes sense in any event to handle override behavior if we'd want it.